syft/frameworks/torch/pointers/object_wrapper.py

```diff
@@ -92,7 +92,7 @@ def create_pointer(
             id_at_location=id_at_location if id_at_location is not None else object.id,
             tags=object.tags,
             description=object.description,
-            garbage_collect_data=garbage_collect_data,
+            garbage_collect_data=False if garbage_collect_data is None else garbage_collect_data,
```

Comment:
Not sure why this is different from the previous implem where the default is also False but stated in the signature

Reply:
The issue is that I have a chain of functions that will end up calling the create_pointer() function. 
And in the upper layers I would like to be able to distinguish between 1) argument not provided => None and 2) argument provided => True/False. And the two separate implementations of create_pointer have different default values for garbage_collect_data.

Reply:
Ok so the main argument would be not to set a default for this boolean value in the signature which differs for 2 functions?